sdk/management/docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,284 @@
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+Azure azure = Azure.authenticate(new File("my.azureauth")).withDefaultSubscription();
+```
+**In new version, this feature has been removed.** If this creates concern on your side, please file an issue to let us know.
+
+For detailed information on the benefits of using the new authentication classes, please refer to [this page](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/management/docs/AUTH.md)
+
+## Customized Policy
+
+Because of adopting Azure Core which is a shared library across all Azure SDKs, there is also a minor change regarding how customized policy in configured. 
+
+In old version (`com.microsoft.azure-mgmt-**`), we use `withInterceptor` and pass the customized interceptor class to the Azure object 
+
+In new version (`com.azure.resourcemanager.**`), we use `WithPolicy` instead and pass the customized policy to the Azure object 
+
+So:
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+Azure azure = Azure.configure()
+    .withInterceptor(new CustomizedInterceptor())
+    .authenticate(credential)
+    .withDefaultSubscription();
+```
+
+**Equivalent in new version (`com.azure.resourcemanager.**`)**
+
+```java
+Azure azure = Azure.configure()
+    .withPolicy(new CustomizedPolicy())
+    .authenticate(credential, profile)
+    .withDefaultSubscription();
+```
+
+## Custom HTTP Client
+
+Similar to the customized policy, there are changes regarding how the custom HTTP client is configured as well. The re-designed HTTP client builder in the new version is more flexible and the user can choose their own implementation of HTTP client and plug in what they need into the configuration.
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+OkHttpClient.Builder builder = new OkHttpClient.Builder().proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 8888)));
+RestClient client = new RestClient.Builder(builder, new Retrofit.Builder())
+    .withCredentials(credential)
+    .build();
+
+Azure azure = Azure.authenticate(client, "<TenantId>")
+    .withDefaultSubscription();
+```
+
+**Equivalent in new version (`com.azure.resourcemanager.**`)**
+
+```java
+HttpClient client = new OkHttpAsyncHttpClientBuilder()
+    .proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress("127.0.0.1", 8888)))
+    .build();
+
+Azure azure = Azure.configure()
+    .withHttpClient(client)
+    .authenticate(credential, profile)
+    .withDefaultSubscription();
+```    
+    
+## Error Handling
+
+There is a minor namespace change in the exception class. To be specific, the previous `CloudException` has been re-named to `ManagementException`. 
+
+**In old version (`com.microsoft.azure-mgmt-**`)**
+
+```java
+final String resourceGroupName = random(8);
+try {
+    ResourceGroup resourceGroup = azure.resourceGroups().getByName(resourceGroupName);
```

Comment:
There could be an issue here.

I think track1 will not have exception on this. There is a kind of hack in track1 runtime lib that will convert 404 to an empty response. So it will give a `null` instead of exception.

Track2 does not do this hack. So it will be an exception.

Almost all service will allow 200 for get method, 404 as error.
However there is one service out there that doing things a bit differently. Appservice would actually allow 404 in their `get` method.

Reply:
For one only need to get an exception, I think one can create a resource group with space in name, e.g. "rg not valid".

Reply:
Oh, yes. I think it could be better.

Reply:
@nickzhums updated in sample

Reply:
this sounds more than just namespace change - maybe we could use a different example? or does this actually cause problem for customer?

Reply:
It is a change on what is an error.

Track1 has a hack that if it is GET http method, and response is 404, it returns empty, but not take it as exception, even service specifies that 404 is error.

Track2 removed this (or, not adding this hack).

Reply:
got it, can we use a different example then, e..g. creating a resource group that has invalid namespace or something like that

Reply:
Tanyi already updated the sample:
https://github.com/Azure-Samples/java-management/blob/migration-guide/src/main/java/com/azure/resourcemanager/samples/ResourceManager.java#L84-L93